sdk/eventhub/azure-eventhub/CHANGELOG.md

```diff
@@ -8,6 +8,8 @@
 
 ### Bugs Fixed
 
+- Fixed a bug that `EventHubProduerClient` is not reusable after getting closed (issue #21849).
```

Comment:
can you explain what you mean by not reusable currently? (just for my own sake, sorry - I read through the issue and the code, but I don't fully understand what the bug is).

Reply:
it is the case when the same client which is opened, closed, then opened again.

```python
producer = EventHubProducerClient(...)
with producer:
   #...
   # the context manager will close the underlying amqp connection

# error would be raised if I want to reuse this producer
with producer:  # the same producer can't work because of the bug when retrieving internal partition producers
```